scroll-animations-1/Overview.bs

```diff
@@ -648,35 +653,72 @@ The procedure to calculate <dfn>effective scroll range</dfn> of a
 
     </div>
 
-### The current time of a {{ScrollTimeline}} ### {#current-time-algorithm}
 
-The [=current time=] of a {{ScrollTimeline}} is calculated as follows:
+### The phase of a {{ScrollTimeline}} ### {#phase-algorithm}
 
-1.  If {{source}} is null, does not currently have a [=CSS layout box=], or if
-    its layout box is not a [=scroll container=], return an unresolved time
-    value.
+The [=phase=] of a {{ScrollTimeline}} is calculated from the first matching
+condition from below:
+
+<div class="switch">
+
+:   If <em>any</em> of the following are true:
+
+    1.  {{source}} is null, or
+    1.  {{source}} does not currently have a [=CSS layout box=], or
+    1.  {{source}}'s layout box is not a [=scroll container=], or
+    1.  [=effective scroll range=] is null
 
-1.  Otherwise, let <var>current scroll offset</var> be the current scroll offset
+::  return [=timeline inactive phase|inactive=]
+
+:   Otherwise, let <var>current scroll offset</var> be the current scroll offset
     of {{source}} in the direction specified by {{orientation}}.
 
-1.  If [=effective scroll range=] is null, return an unresolved time value.
+:   If <var>current scroll offset</var> is less than [=effective start offset=]:
+::  return [=before phase=]
+
+:   If <var>current scroll offset</var> is greater than or equal to [=effective
+    end offset=] <em>and</em> [=effective end offset=] is less than the maximum
+    scroll offset of {{source}} in {{orientation}}:
+::  return [=after phase=]
 
-1.  If <var>current scroll offset</var> is less than [=effective start offset=],
-    return 0.
+:   Otherwise,
+::  return [=active phase=].
 
-    Issue(4325): Define what the correct timeline state is based on scroll
-    offset.
+</div>
 
-1.  If <var>current scroll offset</var> is greater than or equal to [=effective
-    end offset=], return [=effective time range=].
+### The current time of a {{ScrollTimeline}} ### {#current-time-algorithm}
 
-1.  Return the result of evaluating the following expression:
+The [=current time=] of a {{ScrollTimeline}} is calculated from the first
+matching condition from below:
+
+<div class="switch">
```

Comment:
Similarly, I think this could be reworked to be easier to follow using a similar approach to the phase section.